util/autogen_testutils/gen.py

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/env python3
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+import glob
+import logging
+import shutil
+import subprocess
+import sys
+from pathlib import Path
+from typing import List
+
+from autogen_banner import get_autogen_banner
+from make_new_dif.ip import Ip
+from mako.template import Template
+
+# This file is $REPO_TOP/util/autogen_testutils.py, so it takes two parent()
+# calls to get back to the top.
+REPO_TOP = Path(__file__).resolve().parent.parent.parent
+
+
+def gen_testutils(ips_with_difs: List[Ip]) -> None:
+    """Generate testutils libraries that are rendered from Mako templates.
+
+    Args:
+        ips_with_difs: List of IP objects that have existing DIF libraries.
+
+    Returns:
+        None
+    """
+    # Check clang-format is installed.
+    assert (shutil.which("clang-format") and
+            "ERROR: clang-format must be installed to format "
+            " autogenerated code to pass OpenTitan CI checks.")
+
+    # Define input/output directories.
+    testutils_templates_dir = REPO_TOP / "util/autogen_testutils/templates"
+    autogen_testutils_dir = REPO_TOP / "sw/device/lib/testing/autogen"
+
+    # Create output directories if needed.
+    autogen_testutils_dir.mkdir(exist_ok=True)
+
+    # Render templates.
+    for suffix in [".h", ".c", ".build"]:
+        comment_syntax = "//"
+        if suffix == ".build":
+            comment_syntax = "#"
+        for testutils_template_path_str in glob.iglob(
+                str(testutils_templates_dir / f"*{suffix}.tpl")):
```

Comment:
I think iterdir + `if suffix == .tpl: continue` is cleaner.

Reply:
Fixed.